byterun/win32.c

```diff
@@ -731,7 +731,43 @@ int caml_snprintf(char * buf, size_t size, const char * format, ...)
 wchar_t *caml_secure_getenv (wchar_t const *var)
 {
   /* Win32 doesn't have a notion of setuid bit, so getenv is safe. */
-  return CAML_SYS_GETENV (var);
+  return _wgetenv(var);
```

Comment:
Why not calling `caml_win32_getenv` here?  This introduces an inconsistency between the secure and the plain versions of getenv, while I thought the consensus is that under Windows they are identical.

Reply:
The point is that the string returned by `caml_win32_getenv` needs to be explicitly deallocated, as opposed to the one returned by `_wgetenv`. If we want to use `caml_win32_getenv` for `caml_secure_getenv` we need to change all the call sites of `caml_secure_getenv` to require explicit deallocation of the string. This is what the first version of this PR did, but it was a more invasive change. Rather, in the current version, we only use `caml_win32_getenv` for the library functions `{Sys,Unix}.getenv` and the runtime, which uses `caml_secure_getenv`, is left unchanged.

As far as I can see there is no inconsistency between secure and plain versions.  Rather the inconsistency is between the stdlib versions (`caml_sys_getenv`, `caml_sys_unsafe_getenv`, both using `caml_win32_getenv`) and the one used by the runtime (`caml_secure_getenv`, uses `_wgetenv` and there is no separate plain version). We can remove this inconsistency if we shift to an API requiring explicit deallocation of the returned string everywhere.

Reply:
OK, thanks for the explanation.  To help other readers of this code, could you add a comment summarizing this discussion?  E.g. :

caml_win32_getenv is used to implement Sys.getenv and Unix.getenv in such a way that they get direct access to the Win32 environment rather than to the copy that is cached by the C runtime system.  The result of caml_win32_getenv is dynamically allocated and must be explicitly deallocated.

In contrast, the OCaml runtime system still calls _wgetenv from the C runtime system, directly or through caml_secure_getenv.  The result is statically allocated and needs no deallocation.

Reply:
Good idea. Added your suggested comment with a minor tweak.